passgen/diceware/src/main/kotlin/dev/msfjarvis/aps/passgen/diceware/DicewarePassphraseGenerator.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2014-2021 The Android Password Store Authors. All Rights Reserved.
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+
+package dev.msfjarvis.aps.passgen.diceware
+
+import java.io.InputStream
+import javax.inject.Inject
+
+/**
+ * Password generator implementing the Diceware passphrase generation mechanism. For detailed
+ * information on how this works, see https://theworld.com/~reinhold/diceware.html.
+ */
+public class DicewarePassphraseGenerator
+@Inject
+constructor(
+  private val die: Die,
+  wordList: InputStream,
+) {
+
+  private val wordMap = WordListParser.parse(wordList)
+
+  /** Generates a passphrase with [wordCount] words. */
+  public fun generatePassphrase(wordCount: Int, separator: Char): String {
+    return StringBuilder()
+      .apply {
+        repeat(wordCount) { idx ->
+          append(wordMap[die.rollMultiple(DIGITS)])
+          if (idx < wordCount - 1) append(separator)
+        }
+      }
+      .toString()
+      .trimEnd()
```

Comment:
nit: can probably use the `stringBuilder` lambda method here instead of `apply`

Reply:
Apparently the name is `buildString`, I've migrated to it.